### PR TITLE
8231865: JFXPanel sends resize event with size 0x0 on HiDPI devices

### PR DESCRIPTION
**Pull request by prsadhuk**
If the scaleFactors used for the current device are not 1.0, a JXFPanel will send a resize event with 0x0 dimensions to the JavaFX scene which can have undesirable effects, which is because the resize pixel buffer is created even for initial size of 9x0 width,height.
Fix is to make sure to prevent resizing the pixel buffer for initial width/height of 0.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8231865](https://bugs.openjdk.org/browse/JDK-8231865): JFXPanel sends resize event with size 0x0 on HiDPI devices (**Bug** - P3)

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1168/head:pull/1168` \
`$ git checkout pull/1168`

Update a local copy of the PR: \
`$ git checkout pull/1168` \
`$ git pull https://git.openjdk.org/jfx.git pull/1168/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1168`

View PR using the GUI difftool: \
`$ git pr show -t 1168`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1168.diff">https://git.openjdk.org/jfx/pull/1168.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1168#issuecomment-1618816951)

### CODE REVIEW

File: modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java [hunk 1]
```diff
@@ -626,6 +626,9 @@ private void updateComponentSize() {
         newScaleFactorY = GraphicsEnvironment.getLocalGraphicsEnvironment().
                           getDefaultScreenDevice().getDefaultConfiguration().
                           getDefaultTransform().getScaleY();
+        if (oldWidth == 0 && oldHeight == 0 && pWidth == 0 && pHeight == 0) {
+            return;
+        }
```
Comment (andy-goryachev-oracle) [inline modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:631]: this change fixes the issue with the provided test code.

a few comments:
1. if I wrap textArea with a BorderPane, I never get width=0 (master branch)
2. would it make sense to add a check to line 611: if((pWidth == 0) && (pHeight == 0))?
3. lines 617, 618 subtract border insets, would that risk making pWidth/pHeight negative (considering code on lines 613, 614)?

Comment (prsadhuk) [reply modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:631]: I guess l618, 619 in PR takes care of pt 2,3 concern..

**Comment by kevinrushforth** [comment]
Ideally, a regression test -- either automated (if feasible) or manual -- would have been included with this fix. Is it worth a follow-on bug to add such a test?

**Pull request closed without merging**